Add MintingMode::Acl

gRoussac opened the pull request.

gRoussac committed:
Add MintingMode::Acl

gRoussac committed:
Wording

gRoussac committed:
Revert change

gRoussac committed:
Add some tests

gRoussac committed:
linting

gRoussac committed:
small change

gRoussac committed:
move migrating white list to utils

gRoussac committed:
Modify get_verified_caller and acl whitelist to handle whitelisting package

gRoussac committed:
naming tests

gRoussac committed:
small change

deuszex reviewed (approved):
Looks good to me but in the readme and release notes we will have to emphasise the deprecation of contract_whitelist and that acl takes its place.

gRoussac committed:
Update readme + add tests

gRoussac committed:
Test CI/CD

gRoussac committed:
Add test

gRoussac committed:
small change

gRoussac committed:
Naming ACL

gRoussac committed:
Update README.md

Co-authored-by: Adam Stone <97986246+ACStoneCL@users.noreply.github.com>

gRoussac committed:
Update README.md

Co-authored-by: Adam Stone <97986246+ACStoneCL@users.noreply.github.com>

gRoussac committed:
Update README.md

Co-authored-by: Adam Stone <97986246+ACStoneCL@users.noreply.github.com>

gRoussac committed:
Update README.md

Co-authored-by: Adam Stone <97986246+ACStoneCL@users.noreply.github.com>

gRoussac committed:
Update README.md

Co-authored-by: Adam Stone <97986246+ACStoneCL@users.noreply.github.com>

gRoussac committed:
Update README.md

Co-authored-by: Adam Stone <97986246+ACStoneCL@users.noreply.github.com>

gRoussac committed:
Update README.md

Co-authored-by: Adam Stone <97986246+ACStoneCL@users.noreply.github.com>

gRoussac committed:
Update README.md

Co-authored-by: Adam Stone <97986246+ACStoneCL@users.noreply.github.com>

gRoussac committed:
Naming

gRoussac committed:
Small change

gRoussac committed:
Check pakage first on acl contract

gRoussac committed:
Revert packages changes

gRoussac committed:
Rewrite if statement

gRoussac committed:
true/false is some

gRoussac committed:
Update Casper crates

# Conflicts:
#	client/is_approved_for_all_session/Cargo.toml
#	contract/Cargo.toml
#	tests/src/installer.rs
#	tests/src/mint.rs

gRoussac committed:
Linting

gRoussac committed:
serde-json-wasm version

gRoussac committed:
Feature 1.4 update from dev (#226)

* Remove revert ContractAlreadyMigrated (#218)

Add four tests that do

1.2 in OwnerReverseLookupMode::Complete and  OwnerReverseLookupMode::NoLookUp > 1.3

1.0 > 1.2 > 1.3

1.0 > 1.3

Modify test for error 126 to only be raised by 1.2

---------

Co-authored-by: Karan Dhareshwar <42871449+darthsiroftardis@users.noreply.github.com>

* Clarify: OwnerReverseLookupMode must be Complete

* Updated image which was incorrect

* Add an example of owner_reverse_lookup_mode

* Point the README to the tutorials

* Simplify links

* Add clarification for the hash_by_index dictionary

* Minor update

* Review feedback

* Update README.md (#214)

* Update README.md

* Update README.md

Co-authored-by: Iulia Popescu <ipopescu@users.noreply.github.com>

---------

Co-authored-by: Gregory Roussac <gRoussac@users.noreply.github.com>
Co-authored-by: Iulia Popescu <ipopescu@users.noreply.github.com>

* chore: versions bump (#219)

Co-authored-by: Gregory Roussac <gRoussac@users.noreply.github.com>

* Update CHANGELOG.md for 1.3

* Version bump to 1.3

---------

Co-authored-by: Karan Dhareshwar <42871449+darthsiroftardis@users.noreply.github.com>
Co-authored-by: ipopescu <ipopescu@users.noreply.github.com>
Co-authored-by: Jonas Pauli <49498646+jonas089@users.noreply.github.com>
Co-authored-by: Jan Hoffmann <jan@hfmn.pl>
Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>

gRoussac committed:
Merge branch 'feature-1.4' of github.com:casper-ecosystem/cep-78-enhanced-nft into acl

# Conflicts:
#	README.md
#	client/balance_of_session/Cargo.toml
#	client/get_approved_session/Cargo.toml
#	client/is_approved_for_all_session/Cargo.toml
#	client/mint_session/Cargo.toml
#	client/owner_of_session/Cargo.toml
#	client/transfer_session/Cargo.toml
#	client/updated_receipts/Cargo.toml
#	contract/Cargo.toml
#	contract/src/main.rs
#	tests/Cargo.toml

gRoussac committed:
Update test
